src/DynamoCoreWpf/ViewModels/Core/StateMachine.cs

```diff
@@ -947,7 +947,8 @@ internal void PasteSelection(Point2D targetPoint)
                 var oldClipboardData = model.ClipBoard.ToList();
 
                 model.Copy();
-                if (model.ClipBoard.Any())
+                // Prevents Paste from being called when only ConnectorPins are selected.
+                if (model.ClipBoard.Where(m => !(m is ConnectorPinModel)).Select(m => m).Any())
```

Comment:
same here, maybe use `All()`?

Reply:
Here doing so changes the functionality ever so slightly. If we apply `All()` here, whenever there is a pin in the selection, nothing drags. If we keep the old conditional whatever else that isn't a pin copies over. What do you prefer? @QilongTang

Reply:
Hey @M-JULIANI I was only reading your comments and think maybe use the following, what do you think?
```
if (!ClipBoard.All(m => m is ConnectorPinModel))
```

Reply:
```
     if (ClipBoard.All(m => m is ConnectorPinModel))
            {
                return;
            }
```

Reply:
^that is what we need to avoid having pins pasted, which throws an error.. However using this same logic at `StateMachine` prevents other nodes in the selection form being copied as well. Is that expected behaviour or do you want other things in selection to copy over? If that is the case we need to leave the conditional alone at `StateMachine`. @QilongTang

Reply:
All I am saying is this comment has to be discarded if we want other things to be copied (and not pins) when pins are part of the selection. @QilongTang

Reply:
Gotcha! @Amoursol  @Jingyi-Wen any preference?

Reply:
I also think we should have other things copied as well

Reply:
Agreed - please copy all the things :)

Reply:
Thanks! @M-JULIANI Please go ahead with this decision, team will re-test after this PR merged.

Reply:
Sound good, comment resolved! @QilongTang

Reply:
Hey @M-JULIANI I am still not understanding here.. Why dont we use the simplified version since you only mean as long as the `ClipBoard` does not only contain pins, we should proceed right?
```
if (!ClipBoard.All(m => m is ConnectorPinModel))
{
...
}
```

Reply:
Fair enough, just tested it and it's equivalent :)

Reply:
Was confused because in the other block it was the opposite. Fixes made.